### PR TITLE
DOC: make 2x versions of all gallery figures

### PR DESCRIPTION
**Pull request by jklymak**
## PR Summary

This adds the `image_srcset` parameter to our doc conf.py  This will require us to use sphinx-gallery dev version, which I'm not exactly sure how to setup on circleCI.  

https://github.com/sphinx-gallery/sphinx-gallery/pull/808

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Comment by jklymak** [comment]
oooh, that worked.

I guess the issue is if we want to force devs to download off git. 

Both those builds took quite a while (23 minutes and 30 minutes).  I suspect its a time of day thing: a quick survey of a couple of other PRs range from 18-24 minutes, but I'm not going to guarantee this doesn't slow things down.

Compare:

## New

https://58003-1385122-gh.circle-artifacts.com/0/doc/build/html/gallery/lines_bars_and_markers/barh.html

## Old:

https://matplotlib.org/stable/gallery/lines_bars_and_markers/barh.html

**Comment by timhoffm** [comment]
If we want to push this to the test, we'll have to run the new code.

Maybe include a switch so that it only runs on `ci/circleci: docs-python38` and not on `ci/circleci: docs-python37`?

**Comment by jklymak** [comment]
> Maybe include a switch so that it only runs on `ci/circleci: docs-python38` and not on `ci/circleci: docs-python37`?

Right now circle CI just pip installs from  `requirements/doc/doc-requirements.txt`.  I don't know how to ask it to install just one package differently.  I guess we could have two doc-requirements...?

**Comment by timhoffm** [comment]
Use the prerelease for both. It’s sufficient to make the 2x distinction in conf.py.

**Comment by jklymak** [comment]
So, leave it as-is?

File: doc/conf.py [hunk 1]
```diff
@@ -168,6 +168,7 @@ def _check_dependencies():
     'thumbnail_size': (320, 224),
     'compress_images': ('thumbnails', 'images'),
     'matplotlib_animations': True,
+    'image_srcset': ["2x"],
```
Comment (timhoffm) [inline doc/conf.py:171]: Maybe
```suggestion
    # 3.7 CI doc build should not use hidpi images in the testing phase
    'image_srcset': [] if platform.python_version[3:] == '3.7' else ["2x"],
```
That way we have builds with and without hidpi images (ok still using the same code but heh).

Comment (jklymak) [reply doc/conf.py:171]: Oh, I see.  Sure we can try that.

Comment (QuLogic) [reply doc/conf.py:171]: You have `sys` already though: `sys.version_info[:2] == (3, 7)`

**Comment by jklymak** [comment]
3.7: https://58146-1385122-gh.circle-artifacts.com/0/doc/build/html/gallery/lines_bars_and_markers/simple_plot.html
3.8: https://58145-1385122-gh.circle-artifacts.com/0/doc/build/html/gallery/lines_bars_and_markers/simple_plot.html

**Review by timhoffm — APPROVED** [review]

**Review by QuLogic — APPROVED** [review]

**Pull request merged**